compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1666,6 +1666,13 @@ object messages {
     val explanation = ""
   }
 
+  case class ValueClassParameterMayNotBeCallByName(valueClass: Symbol, param: Symbol)(implicit ctx: Context)
```

Comment:
We should reuse `VarValParametersMayNotBeCallByName` instead and maybe addapt a bit it's explanation.

Reply:
That was my first idea, but I thought it might be confusing, because it's not really a `val` and doesn't seem to behave like a case class parameter, which implicitly is a `val`, either.

Reply:
You are right